provider/provider.go

```diff
@@ -69,27 +67,40 @@ func NewProvider(url string, setters ...Option) *Provider {
 	}
 
 	transport := http.NewTransport(url)
-
-	monitor := &Provider{
-		Blockchain: scorumgo.NewClient(transport),
-		Options:    args,
+	return  &Provider{
+		client:  scorumgo.NewClient(transport),
+		Options: args,
 	}
-
-	return monitor
 }
 
-func (p *Provider) Provide(from uint32, eventTypes []event.Type, buffer int) (<-chan event.Event, <-chan error) {
-	c := make(chan event.Event, buffer)
-	e := make(chan error, 1)
-
+func (p *Provider) Provide(from uint32, eventTypes []event.Type, onEvent func(event.Event, error))  {
```

Comment:
it would be nice to add  context for canceling

Reply:
Good point, added